marketplace produce card

he-is-james opened the pull request:
produce cards designed to look like the figma
calls data from the airtable and displays them as produce cards on markeplace
started sending data to produce card, need more props for produce detail screen

he-is-james committed:
initial creation of produce card

allowed pass through of components for name and price, will work on getting test img

he-is-james committed:
added more prop fields

he-is-james committed:
working navigation of card to details

he-is-james committed:
props passed from card to detail screen

need to figure out prop types because of eslint error, still builds though

he-is-james committed:
started styling produce card

need to add icon and figure out flex to position everything

he-is-james committed:
started airtable integration

can call the table to get produce, will need a way to map the data in produce cards

he-is-james committed:
Merge branch 'master' into marketplace

he-is-james committed:
display airtable data as produce cards

needs to fix initial load with useState delay

he-is-james committed:
fixed initial render and styled produce cards

going to marketplace now loads all the produce from the airtable
styled the produce cards to look like the figma (may need to adjust sizing because they are pixels not percentages)

he-is-james committed:
fixed spelling error

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -32,11 +46,51 @@ const styles = StyleSheet.create({
+  const getProduce = () => {
+    const list = [];
+    base('Produce').select({
+      maxRecords: 20,
```
How come you're limiting the number of records?

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -32,11 +46,51 @@ const styles = StyleSheet.create({
+  ));
+
+  useEffect(() => {
+    getProduce();
```
This looks good! How were you able to fix the issue of not loading upon mount?

he-is-james replied:
Unsure, I assume it's how I set up pushing the data to the useState variables

hwang635 commented on src/Marketplace/ProduceCard.js:
```diff
@@ -0,0 +1,94 @@
+    });
+  };
+  return (
+    <TouchableOpacity style={styles.container} onPress={onPressCard}>
```
using TouchableOpacity is a good idea!

hwang635 commented on src/Marketplace/ProduceCard.js:
```diff
@@ -0,0 +1,94 @@
+            {price}
+          </Text>
+          <Text style={styles.quantity}>
+            {' '}
```
Replace the space with more margin/padding?

hwang635 commented on src/Marketplace/ProduceDetailsScreen.js:
```diff
@@ -0,0 +1,30 @@
+})
+
+function ProduceDetailsScreen({ route }) {
+  const { image, name, price } = route.params; // need help figuring out how to get the prop types
```
label comments like these as TODO: [thing to do] for future ref. for prop types, we can go with the general field type in airtable like number/string/function/etc

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -32,11 +46,51 @@ const styles = StyleSheet.create({
+    <ProduceCard
+      style={styles.produceCard}
+      key={produce.Name}
+      navigation={navigation}
```
Can you add checks for null values or some default value for the fields pulled from Airtable?

hwang635 commented on src/Marketplace/ProduceCard.js:
```diff
@@ -0,0 +1,94 @@
+      </View>
+    </TouchableOpacity>
+  );
+}
```
Could you add a little more space between the bottom of the image and start of the text?

hwang635 reviewed (changes requested):
This looks really great! The produce card looks amazing and works really well. Only thing is that I don't think the heart thing does anything right now -- could you add a filled/not filled toggle functionality for the heart if pressed? I left a few questions and comments but otherwise this is perfect.

he-is-james committed:
adjustments to design and render

adjusted checks when pulling from airtable
adjusted design of the produce card
implemented heart like system

he-is-james committed:
pass in correct images and proptypes

hwang635 reviewed (approved):
Thanks for your hard work James!
